### ISSUE TITLE
Refactor subsystem builder methods that can be properties

### ISSUE THREAD

**Issue by johnjasa** [OpenMDAO/Aviary#347]
Multiple different methods in `SubsystemBuilderBase` could be properties to simplify their use. This is backwards incompatible though generally worthwhile; could deprecate the existing methods.

```suggestion
    @property
    def get_mass_names(self):
```
so it can be accessed as `BatteryBuilder.get_mass_names` instead of `BatteryBuilder.get_mass_names()`, since no arguments are used

_Originally posted by @crecine in https://github.com/OpenMDAO/Aviary/pull/317#discussion_r1651562417_

**Comment by crecine** [comment]
I don't think it is backwards incompatible, I'm pretty sure it still works with the parenthesis, they just aren't necessary anymore.

**Comment by Kenneth-T-Moore** [comment]
It might be better to go the other direction and give all of these methods a standard set of arguments that include aviary_inputs, phase name, and phase_info.

**Comment by jkirk5** [comment]
More arguments is probably best, since I'm sure users will have all kinds of odd subsystems that configure themselves based on different things. Having a subsystem need to know the value of specific inputs, or information on the mission is pretty reasonable and we should be able to provide that

**Comment by jkirk5** [comment]
I think making methods into properties is not the best way to go because it limits user options. Instead, we can come up with the list of desired "standardized" arguments for all subsystembuilder methods as suggested by Ken. They should all also support kwargs just to remain flexible.
